packages/core/src/auth/sso/ssoAccessTokenProvider.ts

```diff
@@ -184,19 +184,31 @@ export class SsoAccessTokenProvider {
             clientSecret: registration.clientSecret,
         })
 
-        if (!(await openSsoPortalLink(this.profile.startUrl, authorization))) {
-            throw new CancellationError('user')
+        const openBrowserAndWaitUntilComplete = async () => {
+            if (!(await openSsoPortalLink(this.profile.startUrl, authorization))) {
+                throw new CancellationError('user')
+            }
+
+            const tokenRequest = {
+                clientId: registration.clientId,
+                clientSecret: registration.clientSecret,
+                deviceCode: authorization.deviceCode,
+                grantType: deviceGrantType,
+            }
+
+            return await pollForTokenWithProgress(() => this.oidc.createToken(tokenRequest), authorization)
         }
 
-        const tokenRequest = {
-            clientId: registration.clientId,
-            clientSecret: registration.clientSecret,
-            deviceCode: authorization.deviceCode,
-            grantType: deviceGrantType,
+        let token: ReturnType<typeof openBrowserAndWaitUntilComplete>
+        if (telemetry.spans.map(s => s.name).includes('aws_loginWithBrowser')) {
+            // During certain flows, eg reauthentication, we are already running within a span (run())
+            // so we don't need to create a new one.
```

Comment:
Extending your dedup work in https://github.com/aws/aws-toolkit-vscode/pull/3994 , we could make this happen automatically for various metrics (such as `aws_loginWithBrowser`).

Actually, it's probably rare that we would want to nest a metric in *itself*.